### PR TITLE
Button block: add color support via block.json

### PR DESCRIPTION
**Pull request by oandregal**
Part of https://github.com/WordPress/gutenberg/issues/28913
Depends on https://github.com/WordPress/gutenberg/pull/29378

This PR removes the ad-hoc color support for the block and uses recently added features to the block support mechanism https://github.com/WordPress/gutenberg/issues/28913 that allows us to target elements other than the block wrapper.

## How has this been tested?

This PR depends on https://github.com/WordPress/gutenberg/pull/29378 so, you'd need to apply those changes to your local branch. For example, check out 29378 locally and then `git merge --squash <local-name-of-29378-branch>`.

- Use a theme with theme.json support, such as TT1-blocks.
- Via theme.json target the button color styles by adding the following within the styles section:

```json
"core/button": {
    "color": {
        "text": "green",
        "background": "yellow"
    }
}
```
- Go to the editor, add three buttons: one without user styles, one that uses preset colors, and one that uses custom colors.
- Publish the post and go to the front end.

The expected result is that the button without user styles has the colors declared via theme.json and the other button have the user style colors.

### CODE REVIEW

File: packages/block-library/src/button/block.json [hunk 1]
```diff
@@ -61,8 +61,11 @@
 		"anchor": true,
 		"align": true,
 		"alignWide": false,
+		"color": {
+			"__experimentalSkipSerialization": true
+		},
 		"reusable": false,
-		"__experimentalSelector": ".wp-block-button > a"
+		"__experimentalSelector": ".wp-block-button__link"
```
Comment (nosolosw) [inline packages/block-library/src/button/block.json:68]: It seems that we added the block selector in this PR https://github.com/WordPress/gutenberg/pull/25768 although I haven't it found in use anywhere, so I've reduced its specificity to the minimum.

Note that if we wanted to use the existing selector we'd need to fix https://github.com/WordPress/gutenberg/issues/29381 first.

Comment (youknowriad) [reply packages/block-library/src/button/block.json:68]: I don't have objections here, the simpler the selector the better it seems.

**Review by youknowriad — APPROVED** [review]
This looks good. Love all the removals :) more red please

**Review by aristath — APPROVED** [review]
Nice! LGTM 👍

**Comment by nosolosw** [comment]
Actually, looking at the mobile tests I've noticed we can't remove `ColorEdit` just yet, so it's a bit less of red in this PR. Going to revert that removal, merged this PR afterward, and ask the mobile team how to make those changes for mobile as well (looked at it but wasn't sure how, how to test, etc).

**Comment by geriux** [comment]
> I reached out to @chipsnyder and @geriux for feedback on this.

Hey 👋 I'm going to test this now on mobile and see if it's all working correctly.

**Review by geriux — APPROVED** [review]
All good from the mobile side ✅ Thanks for reaching out to give it a quick test!

**Pull request merged**